etna/models/prophet.py

```diff
@@ -152,6 +154,76 @@ def predict(self, df: pd.DataFrame, prediction_interval: bool, quantiles: Sequen
         y_pred = y_pred.rename(rename_dict, axis=1)
         return y_pred
 
+    def _check_mul_components(self, ignore_components: Set[str]):
+        """Raise error if model contains multiplicative components."""
+        components_modes = self.model.component_modes
+        if components_modes is None:
+            raise ValueError("This model is not fitted!")
+
+        mul_components = set(self.model.component_modes["multiplicative"])
+        if len(mul_components - ignore_components) > 0:
+            raise ValueError("Forecast decomposition is only supported for additive components!")
+
+    def _predict_seasonal_components(self, df: pd.DataFrame, ignore_components: Set[str]) -> pd.DataFrame:
+        """Estimate seasonal, holidays and exogenous components."""
+        model = self.model
+
+        seasonal_features, _, component_cols, _ = model.make_all_seasonality_features(df)
+
+        holiday_names = set(model.train_holiday_names) if model.train_holiday_names is not None else set()
+
+        components_data = {}
```

Comment:
We can try to do here vecorization, but main reason of this solution is to persist consistency with similar [method](https://github.com/facebook/prophet/blob/514a5a29b4d8878f8873ab57dd6be32ee8ec4a47/python/prophet/forecaster.py#L1341) from `prophet`. I don't think we will gain much speed up vectorizing this cycle. Also, we can't throw out the cycle fully as we need to filter individual holidays and aggregated components from columns in `component cols`.

Reply:
First of all, we can filter out the unnecessary components before the cycle, the second thing is that Prophet team does not plan to change the model, so we don't need to maintain code consistency if it can b vectorized